pkg/k8s/portforward.go

```diff
@@ -56,7 +58,9 @@ func NewPortForward(
 		return nil, err
 	}
 
-	pods, err := kubeAPI.GetPodsByNamespace(client, namespace)
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	pods, err := kubeAPI.GetPodsByNamespace(ctx, client, namespace)
```

Comment:
Just curious about the choice of a 5-second timeout for this call when setting up the port-forward. Maybe it should be 30 seconds as well? Before we get to this code when running `linkerd dashboard`, we've already made other API calls, all with a 30 second timeout. If we're trying to talk to a cluster with really slow response times, my guess is that we would need a 30 second timeout here.

Reply:
I only changed the timeouts relevant to the checks, but yeah, I agree we can have this one be 30s as well.